source/Cosmos.HAL2/BlockDevice/AHCI.cs

```diff
@@ -268,6 +271,7 @@ private PortType CheckPortType(PortRegisters aPort)
                 case AHCISignature.SEMB: return PortType.SEMB;
                 case AHCISignature.PortMultiplier: return PortType.PM;
                 case AHCISignature.Nothing: return PortType.Nothing;
+                default: return (PortType)0x7FFF;
```

Comment:
Can you please "indent" the switch correctly with the return on the next line?
I don't undestand why you are not throwing in default of the switch instead of setting the PortType to 0x7FFF?